libs/memoryApi/include/memoryApi/memoryComparison.h

```diff
@@ -0,0 +1,62 @@
+/**
+  Include this file to get templetized memory based comparisons, including
+  templetized Functors which are handy with STL containers.
+*/
+
+#pragma once
+#ifndef MSO_PLATFORMADAPTERS_MEMORY_COMPARISON_H
+#define MSO_PLATFORMADAPTERS_MEMORY_COMPARISON_H
+
+#include <memory>
+
+namespace Mso::Memory {
+template <typename T>
+int Compare(const T& lhs, const T& rhs) noexcept

```

Comment:
>Compare [](http://example.com/codeflow?start=4&length=7)

These are templates - they are not even compiled until we use them. Thus, all new templates must have unit tests. Even if they are trivial - they are still good to see that code at least compiles.